### DISCUSSION TITLE
ESP32 Ethernet Support

### DISCUSSION THREAD

**Discussion by maweicoltsmart** [zephyrproject-rtos/zephyr#35801] [Q&A]
Is anyone working on porting the Ethernet driver of ESP32? By the way, tell me the address of the git repository, thank you!

**Comment by jukkar** [comment]
cc: @sylvioalves

**Comment by nashif** [comment]
did not know ESP32 has ethernet, only Wifi

**Reply by sylvioalves** [reply]
Yes, there is: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/hw-reference/esp32/get-started-ethernet-kit.html

**Reply by maweicoltsmart** [reply]
Some ESP32 sub-models support Ethernet, you can get this information at https://products.espressif.com/#/product-selector?language=zh.
It is useful when using ESP32 as an IoT gateway. So, are there any plans to surport Ethernet drivers?

**Reply by maweicoltsmart** [reply]
Can you reopen this issue?

**Reply by nashif** [reply]
what issue, this issue is still open.

**Reply by maweicoltsmart** [reply]
Do you plan to migrate the ESP32 Ethernet driver? When is it expected?

**Reply by sylvioalves** [reply]
@maweicoltsmart I am evaluating the time frame for this, will ping you soon.

**Comment by maweicoltsmart** [comment]
hi，Are there plans for Ethernet support?

--------------原始邮件--------------
发件人："Sylvio Alves ***@***.***&gt;;
发送时间：2021年6月5日(星期六) 中午11:00
收件人："zephyrproject-rtos/zephyr" ***@***.***&gt;;
抄送："maweicoltsmart ***@***.***&gt;;"Mention ***@***.***&gt;;
主题：Re: [zephyrproject-rtos/zephyr] ESP32 Ethernet Support (#35801)
-----------------------------------

 

 
@maweicoltsmart I am evaluating the time frame for this, will ping you soon.

**Reply by sylvioalves** [reply]
Hi @maweicoltsmart, sorry, no plans yet. Do you think you can contribute with that?
Editing: We have lots of other stuff to bring up before handling ethernet driver, which means working on this by late Q4.

**Comment by maweicoltsmart** [comment]
I tried it and found it was too difficult for me

--------------原始邮件--------------
发件人："Sylvio Alves ***@***.***&gt;;
发送时间：2021年6月18日(星期五) 凌晨1:15
收件人："zephyrproject-rtos/zephyr" ***@***.***&gt;;
抄送："maweicoltsmart ***@***.***&gt;;"Mention ***@***.***&gt;;
主题：Re: [zephyrproject-rtos/zephyr] ESP32 Ethernet Support (#35801)
-----------------------------------

 

 
Hi @maweicoltsmart, sorry, no plans yet. Do you think you can contribute with that?

**Comment by grantramsay** [comment]
Just opened a PR open for this: https://github.com/zephyrproject-rtos/zephyr/pull/50141

**Reply by grantramsay** [reply]
Merged 🎉
